File: ObjC/01-复杂度/01-复杂度/Times.m
//
//  Times.m
//  01-复杂度
//
//  Created by CoderSLZeng on 2019/10/21.
//  Copyright © 2019 Franklin Tsang. All rights reserved.
//

#import "Times.h"

@implementation Times

+ (void)test:(NSString *)title taskBlock:(void (^)(void))block {
    NSLog(@"【%@】", title);
    NSDateFormatter *fmt = [[NSDateFormatter alloc] init];
    fmt.dateFormat = @"HH:mm:ss.SSSS";
    NSDate *beginDate = [[NSDate alloc] init];
    NSLog(@"开始：%@", [fmt stringFromDate:beginDate]);
    NSTimeInterval begin = [beginDate timeIntervalSince1970];
    block();
    NSDate *endDate = [[NSDate alloc] init];
    NSLog(@"结束：%@", [fmt stringFromDate:endDate]);
    NSTimeInterval end = [endDate timeIntervalSince1970];
    NSString *delta = [NSString stringWithFormat:@"%.6f", (end - begin)];
    NSLog(@"耗时：%@", delta);
    NSLog(@"---------------------------------------------------");
}
@end

File: Swift/01-复杂度/01-复杂度/Times.swift
//
//  Times.swift
//  01-复杂度
//
//  Created by CoderSLZeng on 2019/10/21.
//  Copyright © 2019 Franklin Tsang. All rights reserved.
//

import Foundation

struct Times {
          
    static func test(_ title:  String, _ task: () -> Void) {
            
        print("【" + title + "】")
        let fmt = DateFormatter()
        fmt.dateFormat = "HH:mm:ss.SSSS"
        let dateStr = fmt.string(from: Date())
        print("开始：" + dateStr)
        let begin = Date().timeIntervalSince1970
        task()
        let end = Date().timeIntervalSince1970
        let endStr = fmt.string(from: Date())
        print("结束：" + endStr)
        let delta = String(format: "%.6f", (end - begin))
        print("耗时：" + delta )
        print("----------------------------------------")        
    }

    
}
